topics/compiler/index.md

```diff
@@ -0,0 +1,8 @@
+---
+display_name: Compiler
+short_description: Compiler is a program that translates high level language (human readable) to low level language (machine language).
+topic: compiler
+related: interpreter, gcc, fortran
+wikipedia_url: https://en.wikipedia.org/wiki/Compiler
+---
+A compiler is a software that translate high level language to executable machine code in one go, and reports error if any.The processor executes object code, which indicates when binary high and low signals are required in the arithmetic logic unit of the processor. Examples of compilers include BASIC compilers, fortran complier, C++, C, java compiler, etc.
```

Comment:
- `reports errors (if any)`
- `any.The` needs a space
- Capitalise `fortran` and `java`
- `compilers include`...  you can remove the word `compiler` after that